include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -309,6 +321,94 @@ MGConstrainedDoFs::initialize(const DoFHandler<dim, spacedim> &dof)
 }
 
 
+template <int dim, int spacedim>
+inline void
+MGConstrainedDoFs::initialize(
+  const DoFHandler<dim, spacedim> &dof,
+  const MGLevelObject<IndexSet> &  level_relevant_dofs)
```

Comment:
Can we make the other function create an `IndexSet` and then let it use the code here to avoid duplicating 85 lines of code?

Reply:
@masterleinad @kronbichler You are both obviously absolutely right, the code duplication is really nasty -the cause for this is that I normally don't like default arguments for reference parameters and didn't want to introduce smart pointers or something like std::optional.
Nevertheless I have changed the code and unified everything to one function with a second default argument, which basically checks by a call to max_level() whether a second parameter was passed by the caller or the default one has been constructed, which avoids code duplication.

Reply:
You also have added a second function and let the implementation of the old function use the new one, but this is also nice.